src/ngx_http_auth_jwt_module.c

```diff
@@ -177,8 +199,45 @@ static ngx_int_t ngx_http_auth_jwt_handler(ngx_http_request_t *r)
 	else if ( auth_jwt_algorithm.len == sizeof("RS256") - 1 && ngx_strncmp(auth_jwt_algorithm.data, "RS256", sizeof("RS256") - 1) == 0 )
 	{
 		// in this case, 'Binary' is a misnomer, as it is the public key string itself
-		keyBinary = jwtcf->auth_jwt_key.data;
-		keylen = jwtcf->auth_jwt_key.len;
+		ngx_log_error(NGX_LOG_ERR, r->connection->log, 0, "failed to find a jwt");
+		if (jwtcf->auth_jwt_use_keyfile == 1)
+		{
+			FILE *file = fopen((const char*)jwtcf->auth_jwt_keyfile_path.data, "rb");
+
+			// Check if file exists or is correctly opened
+			if (file == NULL)
+			{
+				char err[100];
+				strcpy(err, "failed to open pub key file: ");
+				strcat(err, KEY_FILE_PATH);
+				ngx_log_error(NGX_LOG_ERR, r->connection->log, 0, err);
+				goto redirect;
+			}
+
+			// Read file length
+			fseek(file, 0, SEEK_END);
```

Comment:
This is going to read the public key from a file for every request that NGINX receives.  I'm afraid that this would offer poor performance.  I think we would need to read from a file when the configuration is loaded instead.

Reply:
Until this I've never worked with custom Nginx plugins, it's definitely poor performance as the key won't change frequently (changing the key would require a server restart, which makes sense).

Reply:
I've tried the `init_module` and `postconfiguration` hooks, but it seems they don't load the blocks at that time (since when logging the values from the config it displays the default -1 value). I'll try to put it in the `merge_location` hook, since it gets called once (maybe a better idea?).